flash_handler.c - wrong syscall includes

[flash_handler.c](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/flash/flash_handlers.c) includes [<syscalls/flash_sfdp_read.c>](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/flash/flash_handlers.c#L97) and [<syscalls/flash_sfdp_jedec_id.c>](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/flash/flash_handlers.c#L106).

On "my machine" (NCS 2.5.1 / v3.4.99-ncs1-1) this files are named `<syscalls/flash_sfdp_read_mrsh.c>` and `<syscalls/flash_read_jedec_id_mrsh.c>`.

Building with "CONFIG_USERSPACE=y" fails therefore with:

> zephyr/drivers/flash/flash_handlers.c:97:10: fatal error: syscalls/flash_sfdp_read.c: No such file or directory
   97 | #include <syscalls/flash_sfdp_read.c>

Building with "CONFIG_USERSPACE=n" works.

Changing the filenames to the available "_mrsh.c" ones makes building with "CONFIG_USERSPACE=y" also working.

I guess this are just typos / left overs.

I am able to reproduce... kinda. Looks like this might be a build system or west issue (or even expected behavior, I am really not familiar at all with userspace and syscalls)?

This works:

```
west build -p always -b frdm_k64f samples/userspace/hello_world_user -- -DCONFIG_FLASH=y
```

This does not:

```
west build -p always -b frdm_k64f samples/userspace/hello_world_user
west build -p auto -b frdm_k64f samples/userspace/hello_world_user -- -DCONFIG_FLASH=y

...
-- west build: building application
[58/150] Building C object zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_handlers.c.obj
FAILED: zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_handlers.c.obj 
ccache /Users/kartben/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DCPU_MK64FN1M0VLL12 -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DPICOLIBC_LONG_LONG_PRINTF_SCANF -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I/Users/kartben/zephyrproject/zephyr/include -I/Users/kartben/zephyrproject/zephyr/build/zephyr/include/generated -I/Users/kartben/zephyrproject/zephyr/soc/arm/nxp_kinetis/k6x -I/Users/kartben/zephyrproject/zephyr/soc/arm/nxp_kinetis/common -I/Users/kartben/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include -I/Users/kartben/zephyrproject/zephyr/modules/cmsis/. -I/Users/kartben/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MK64F12 -I/Users/kartben/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MK64F12/drivers -I/Users/kartben/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/common -I/Users/kartben/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/uart -I/Users/kartben/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/flash -I/Users/kartben/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/port -I/Users/kartben/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/sysmpu -I/Users/kartben/zephyrproject/zephyr/modules/hal_nxp/. -isystem /Users/kartben/zephyrproject/zephyr/lib/libc/common/include -Wshadow -fno-strict-aliasing -Os -imacros /Users/kartben/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m4 -mthumb -mabi=aapcs -mfp16-format=ieee -mtp=soft --sysroot=/Users/kartben/zephyr-sdk-0.16.1/arm-zephyr-eabi/arm-zephyr-eabi -imacros /Users/kartben/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/Users/kartben/zephyrproject/zephyr/samples/userspace/hello_world_user=CMAKE_SOURCE_DIR -fmacro-prefix-map=/Users/kartben/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/Users/kartben/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections --specs=picolibc.specs -std=c99 -MD -MT zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_handlers.c.obj -MF zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_handlers.c.obj.d -o zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_handlers.c.obj -c /Users/kartben/zephyrproject/zephyr/drivers/flash/flash_handlers.c
/Users/kartben/zephyrproject/zephyr/drivers/flash/flash_handlers.c:19:10: fatal error: syscalls/flash_read_mrsh.c: No such file or directory
   19 | #include <syscalls/flash_read_mrsh.c>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[71/150] Building C object zephyr/drivers/gpio/CMakeFiles/drivers__gpio.dir/gpio_handlers.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /opt/homebrew/bin/cmake --build /Users/kartben/zephyrproject/zephyr/build
...


```

> zephyr/drivers/flash/flash_handlers.c:19:10: fatal error: syscalls/flash_read_mrsh.c: No such file or directory
   19 | #include <syscalls/flash_read_mrsh.c>

Hm, that may point more to the general question, if the generated files or the includes are wrong. I consider, that the developers need to sync on that topic.

--- a/drivers/flash/flash_handlers.c
+++ b/drivers/flash/flash_handlers.c
@@ -94,7 +94,7 @@ static inline int z_vrfy_flash_sfdp_read(const struct device *dev,
 	K_OOPS(K_SYSCALL_MEMORY_WRITE(data, len));
 	return z_impl_flash_sfdp_read(dev, offset, data, len);
 }
-#include <syscalls/flash_sfdp_read.c>
+#include <syscalls/flash_sfdp_read_mrsh.c>
 
 static inline int z_vrfy_flash_read_jedec_id(const struct device *dev,
 					     uint8_t *id)
@@ -103,7 +103,7 @@ static inline int z_vrfy_flash_read_jedec_id(const struct device *dev,
 	K_OOPS(K_SYSCALL_MEMORY_WRITE(id, 3));
 	return z_impl_flash_read_jedec_id(dev, id);
 }
-#include <syscalls/flash_sfdp_jedec_id.c>
+#include <syscalls/flash_read_jedec_id_mrsh.c>
 
 #endif /* CONFIG_FLASH_JESD216_API */
 

